### PR TITLE
[W5.3b][W10-A1]Joshua Lee Kai Sheng

### PR DESCRIPTION
**Pull request by jlks96**
-Extracted commonalities from Phone, Email and Address classes into a parent class called Contact

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:5]: This class and its non-trivial methods should have header comments. 
If not, developers who write client code for this class will not know how to use it.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+    public final String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:8]: Try to make it private. 
In general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+
+    @Override
+    public int hashCode() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:41]: In general, `equals()` and `hashCode()` methods should be overridden together. Find out why!

**Comment by okkhoy** [comment]
@jlks96 
Some comments added. Please ack & close the PR after reading comments.

**Comment by jlks96** [comment]
Acknowledged. Thanks.

**Pull request closed without merging**